connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala

```diff
@@ -1508,8 +1508,10 @@ class SparkConnectPlanner(val session: SparkSession) {
         maxRecordsPerBatch,
         maxBatchSize,
         timeZoneId)
-      assert(batches.size == 1)
-      batches.next()
+      assert(batches.hasNext)
```

Comment:
Sorry for the late reply but how is this code different to the existing one?

val bytes = batches.next()
bytes


Is the same as

batches.next()


The asserts in between don't count as they don't have side effects.

Reply:
The `batches` is an iterator, so `batches.size` consumes all the data in the iterator to calculate the size.
Then `batches.next()` would return an empty data? Usually it should throw an Exception, though.

Reply:
I see thank you. My understanding with iterators was that non rewindable iterators simply do not have a size method. 

But something new learned.